src/fides/api/alembic/migrations/versions/1e750936a141_migrate_to_fideslang_2_0_uses_and_.py

```diff
@@ -0,0 +1,308 @@
+                {"dataset_id": row["id"], "updated_labels": updated_labels},
+            )
+
+        # Update the collections objects
+        collections: str = json.dumps(row["collections"])
+
+        for key, value in data_label_map.items():
+            collections = collections.replace(key, value)
+
+        update_collections_query: TextClause = text(
+            "UPDATE ctl_datasets SET collections = :updated_collections WHERE id= :dataset_id"
+        )
+        bind.execute(
+            update_collections_query,
+            {"dataset_id": row["id"], "updated_collections": collections},
+        )
+
+
+def update_system_ingress_egress_data_categories(
+    bind: Connection, data_label_map: Dict[str, str]
+) -> None:
+    """Upgrade or downgrade system DataFlow objects"""
+    existing_systems: ResultProxy = bind.execute(
+        text("SELECT id, egress, ingress FROM ctl_systems;")
+    )
+
+    for row in existing_systems:
+        ingress = row["ingress"]
+        egress = row["egress"]
+
+        # Do a blunt find/replace
+        if ingress:
+            for item in ingress:
+                item["data_categories"] = [
+                    label.replace(key, value) if key in label else label
+                    for key, value in data_label_map.items()
+                    for label in item.get("data_categories", [])
+                ]
+
+        if egress:
+            for item in egress:
+                item["data_categories"] = [
+                    label.replace(key, value) if key in label else label
+                    for key, value in data_label_map.items()
+                    for label in item.get("data_categories", [])
+                ]
+
+        if ingress:
+            update_ingress_query: TextClause = text(
+                "UPDATE ctl_systems SET ingress = :updated_ingress WHERE id= :system_id"
+            )
+            bind.execute(
+                update_ingress_query,
+                {"system_id": row["id"], "updated_ingress": json.dumps(ingress)},
+            )
+
+        if egress:
+            update_egress_query: TextClause = text(
+                "UPDATE ctl_systems SET egress = :updated_egress WHERE id= :system_id"
+            )
+            bind.execute(
+                update_egress_query,
+                {"system_id": row["id"], "updated_egress": json.dumps(egress)},
+            )
+
+
+def update_privacy_declaration_data_categories(
+    bind: Connection, data_label_map: Dict[str, str]
+) -> None:
+    """Upgrade or downgrade data uses from fideslang 2.0 for data categories"""
+    existing_privacy_declarations: ResultProxy = bind.execute(
+        text("SELECT id, data_categories FROM privacydeclaration;")
+    )
+    for row in existing_privacy_declarations:
+        labels = [
+            label.replace(key, value) if key in label else label
+            for key, value in data_label_map.items()
+            for label in row["data_categories"]
+        ]
```

Comment:
This is not formulating the replacement categories correctly, 

if row["data_categories"] = `['user', 'user.financial.account_number']`

labels becomes: 
`
['user', 'user.financial.bank_account', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number', 'user', 'user.financial.account_number']
`

I'll look into rewriting these -

Reply:
addressed here https://github.com/ethyca/fides/pull/3933/commits/9308569ad9e9342d9e43256f65a3f9bd266c20cd